modules/storage/src/main/kotlin/edu/stanford/spezi/modules/storage/local/LocalStorage.kt

```diff
@@ -0,0 +1,89 @@
+package edu.stanford.spezi.modules.storage.local
+
+import android.content.Context
+import dagger.hilt.android.qualifiers.ApplicationContext
+import edu.stanford.spezi.modules.storage.secure.SecureStorage
+import java.io.File
+import java.security.Key
+import javax.crypto.Cipher
+import javax.inject.Inject
+import kotlin.reflect.KClass
+
+class LocalStorage @Inject constructor(
+    @ApplicationContext val context: Context,
+) {
+    private val secureStorage = SecureStorage(context)
+
+    private fun createCipher(mode: Int, key: Key): Cipher =
+        // TODO: Supported values: https://developer.android.com/reference/kotlin/javax/crypto/Cipher
+        Cipher.getInstance("RSA/ECB/OAEPWithSHA-1AndMGF1Padding").apply { init(mode, key) }
+
+    fun store(storageKey: String, data: ByteArray, settings: LocalStorageSetting) =
+        store(file(storageKey, ByteArray::class), data, settings)
+
+    fun <C : Any> store(
+        value: C,
+        storageKey: String? = null,
+        type: KClass<C>,
+        settings: LocalStorageSetting,
+        encode: (C) -> ByteArray,
```

Comment:
I see the encode and decode is replicating `JSONEncoder/Decoder` in iOS. The kotlin way to handle this is using `kotlinx.serialization` which is also a dependency of the module already:

```kotlin
fun <C : Any> store(
        key: String,
        value: C,
        serializer: KSerializer<C>,
        settings: LocalStorageSetting,
    ) {
        runCatching {
            val data = Json.encodeToString(serializer, value).toByteArray(StandardCharsets.UTF_8)
            // handle data
        }
    }
```

or we can also derive the serializer internally - requires the function to be inline and C type to be [reified](https://kotlinlang.org/docs/inline-functions.html#reified-type-parameters), it works by default for all primitive types, and complex types should be annotated with `@Serializable`. However lists require explicit handling, see some usage in `KeyValueStorageSerialization.kt` file.

```kotlin
inline fun <reified C : Any> store(
    key: String,
    value: C,
    settings: LocalStorageSetting,
) {
    runCatching {
        val data = Json.encodeToString(serializer<C>(), value).toByteArray(StandardCharsets.UTF_8)
        // handle data
    }
}
inline fun <reified C : Any> storeList(
    key: String,
    value: List<C>,
    settings: LocalStorageSetting,
) {
    runCatching {
        val data = Json.encodeToString(ListSerializer(serializer()), value).toByteArray(StandardCharsets.UTF_8)
        // handle data
    }
}
```

Try catch / runCatching is also important to avoid app crashes. Also I would run operations in this class a background thread (io dispatcher)